scipy/special/_support_alternative_backends.py

```diff
@@ -19,6 +20,9 @@ def get_array_special_func(f_name, xp, n_array_args):
     if is_numpy(xp):
         f = getattr(_ufuncs, f_name, None)
     elif is_torch(xp):
+        # These functions are not implemented.
+        if f_name in {'gamma', 'betaln', 'beta', 'betainc', 'expi', 'expn'}:
+            raise NotImplementedError
```

Comment:
```suggestion
```
I expect them to work on CPU through the numpy round-trip a few lines below.
On GPU, they should be skipped in the test (there is already a section for it).

Reply:
Do we have a way to test for PyTorch-cpu?  If not, does it makes more sense to block them on PyTorch since it would lead to the odd behavior whereby a program works just fine on PyTorch-cpu, but fails on PyTorch-gpu?

Reply:
```bash
python dev.py test -b torch  # on CPU float64
SCIPY_DEVICE=cuda python dev.py test -b torch  # on GPU float64
SCIPY_DEFAULT_DTYPE=float32 python dev.py test -b torch  # on CPU float32
```
CI runs all three of them.

>  a program works just fine on PyTorch-cpu, but fails on PyTorch-gpu?

We already have tons of stuff that works fine on torch/jax CPU but fails on GPU. Basically anything with a Cython or C internal call.

Reply:
Sure, but it seems that you're asking me to let these functions pass through here?  Do you want me to unconditionally let them pass through or just when PyTorch is running on the CPU?

Reply:
unconditionally. They will then crash and burn when `np.asarray` tries to implicitly move the GPU buffer to host memory a few lines later.